ec/ecresource/organizationresource/mapper_to_model.go

```diff
@@ -0,0 +1,194 @@
+				ForAllDeployments: forAllApiToModel(roleAssignment.All),
+				DeploymentIDs:     deploymentIds,
+				ApplicationRoles:  applicationRoles,
+			})
+		}
+	}
+	roleAssignments, diags := types.SetValueFrom(ctx, setElementObjectType(deploymentRoleAssignmentsSchema()), result)
+	if diags.HasError() {
+		diagnostics.Append(diags...)
+		return nil
+	}
+
+	return &roleAssignments
+}
+
+func elasticsearchRolesApiToModel(ctx context.Context, member models.OrganizationMembership, diagnostics *diag.Diagnostics) *types.Set {
+	rolesSchema := projectElasticsearchRolesSchema()
+	if member.RoleAssignments != nil && member.RoleAssignments.Project != nil && member.RoleAssignments.Project.Elasticsearch != nil {
+		return rolesApiToModel(ctx, member.RoleAssignments.Project.Elasticsearch, rolesSchema, "elasticsearch", diagnostics)
+	} else {
+		return emptySet()
+	}
+}
+
+func observabilityRolesApiToModel(ctx context.Context, member models.OrganizationMembership, diagnostics *diag.Diagnostics) *types.Set {
+	rolesSchema := projectObservabilityRolesSchema()
+	if member.RoleAssignments != nil && member.RoleAssignments.Project != nil && member.RoleAssignments.Project.Observability != nil {
+		return rolesApiToModel(ctx, member.RoleAssignments.Project.Observability, rolesSchema, "observability", diagnostics)
+	} else {
+		return emptySet()
+	}
+}
+
+func securityRolesApiToModel(ctx context.Context, member models.OrganizationMembership, diagnostics *diag.Diagnostics) *types.Set {
+	rolesSchema := projectSecurityRolesSchema()
+	if member.RoleAssignments != nil && member.RoleAssignments.Project != nil && member.RoleAssignments.Project.Security != nil {
+		return rolesApiToModel(ctx, member.RoleAssignments.Project.Security, rolesSchema, "security", diagnostics)
+	} else {
+		return emptySet()
+	}
+}
+
+func emptySet() *types.Set {
+	value := types.SetValueMust(projectRoleAssignmentSchema([]string{}).Type(), []attr.Value{})
+	return &value
+}
+
+func rolesApiToModel(
+	ctx context.Context,
+	apiRoleAssignments []*models.ProjectRoleAssignment,
+	schema schema.SetNestedAttribute,
+	roleType RoleType,
+	diagnostics *diag.Diagnostics,
+) *types.Set {
+	var result []ProjectRoleAssignment
+
+	for _, roleAssignment := range apiRoleAssignments {
+		projectIds, diags := types.SetValueFrom(ctx, types.StringType, roleAssignment.ProjectIds)
+		if diags.HasError() {
+			diagnostics.Append(diags...)
+			return nil
+		}
+		applicationRoles, diags := types.SetValueFrom(ctx, types.StringType, roleAssignment.ApplicationRoles)
+		if diags.HasError() {
+			diagnostics.Append(diags...)
+			return nil
+		}
+
+		result = append(result, ProjectRoleAssignment{
+			Role:             types.StringValue(roleApiToModel(*roleAssignment.RoleID, roleType)),
+			ForAllProjects:   forAllApiToModel(roleAssignment.All),
+			ProjectIDs:       projectIds,
+			ApplicationRoles: applicationRoles,
+		})
+	}
+
+	roleAssignments, diags := types.SetValueFrom(ctx, setElementObjectType(schema), result)
```

Comment:
IMO we can simplify this:
```suggestion
	roleAssignments, diags := types.SetValueFrom(ctx, schema.GetNestedObject().GetAttributes().Type(), result)
```

Reply:
Yeah that's simpler :+1: